README.md

```diff
@@ -9,3 +9,664 @@ ExpressRouterAdapter is available as a package on NPM for use with a module bund
+
+```js
+app.use(err: any, req: any, res: any, next: any): void {
+  const {
+    status = 500,
+    message = 'Server Error'
+  } = err;
+  if (status >= 500) {
+    console.log('failed to process request', req.method, req.path);
+    console.error('error handler error', err);
+  }
+
+  let body = err.body || {
+      status,
+      message
+  };
+
+  res.status(status)
+    .json(body);
+}
+
+```
+
+## What if I want more control over the status code or headers
+ ```js
+ import { HTTPResponse, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(async ({ petId }) => {
+          return new HTTPResponse({
+            status: 400,
+            body: {
+              message: `pet was missing name`,
+              code: 123
+            }
+          });
+        })
+    ];
+  }
+}
+```
+
+You also can throw
+```js
+ import { HTTPError, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(async ({ petId }) => {
+          throw new HTTPError({
+            message: `pet validation failed`,
+            status: 400,
+            body: {
+              message: `pet was missing name`,
+              code: 123
+            }
+          });
+        })
+    ];
+  }
+}
+```
+
+Which is convenient as you abstract things away, so you can end up with code like
+```js
+ import { HTTPError, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(async ({ body }) => {
+          validateCreatePet(body);
+          // proceed to create the pet
+        })
+    ];
+  }
+}
+```
+
+
+## Can I have more than one media type on a resource?
+Yes, in fact this is one of the primary motivations behind express-route-adapter is to make that easy.
+
+In previous examples, we aren't specifying the media type, and so we're getting the default json formatter.
+
+Consider this example
+
+```js
+import { IHTTPRoute, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
+
+interface IPet {
+  name: string;
+  internalS3PathToPicture: string;
+}
+
+export class PetRouter {
+  pets: IPet[] = [
+    { name: 'honey', internalS3PathToPicture: '...' },
+    { name: 'poseidon', internalS3PathToPicture: '...' }
+  ];
+  getRoutes(): IHTTPRoute[] {
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .get(({ petId }) => {
+          return this.pets[petId];
+        })
+    ];
+  }
+}
+```
+
+Our data model has an internal path that we don't want to leak out.
```

Comment:
Maybe move this up to the 'consider this example' sentence to give readers this context before they read the code sample?